### ISSUE TITLE
[LUIS] Node.js LUIS intent doesn't handle composite entities

### ISSUE THREAD

**Issue by josephktcheung** [Microsoft/BotBuilder#2610]
Hi,

I've trained LUIS with some composite entities but it seems that it doesn't pass those information to the `intent` in the dialog.

Here's 1 example of the intent we receive in the node.js client:

```json
{
  "action": "LuisLibrary:/setFilter",
  "intent": {
    "score": 0.999961853,
    "intent": "SetFilters",
    "intents": [
      {
        "intent": "SetFilters",
        "score": 0.999961853
      }
    ],
    "entities": [
      {
        "entity": "bedrooms",
        "type": "numericalFilters::minBedrooms",
        "startIndex": 18,
        "endIndex": 25,
        "score": 0.999940455
      },
      {
        "entity": "3 bedrooms",
        "type": "setNumerical",
        "startIndex": 16,
        "endIndex": 25,
        "score": 0.9976944
      },
      {
        "entity": "rent",
        "type": "listingType::rent",
        "startIndex": 11,
        "endIndex": 14,
        "score": 0.9967319
      },
      {
        "entity": "20k",
        "type": "builtin.number",
        "startIndex": 7,
        "endIndex": 9,
        "resolution": {
          "value": "20000"
        }
      },
      {
        "entity": "3",
        "type": "builtin.number",
        "startIndex": 16,
        "endIndex": 16,
        "resolution": {
          "value": "3"
        }
      },
      {
        "entity": "20k rent",
        "type": "setListingPrice",
        "startIndex": 7,
        "endIndex": 14,
        "score": 0.987298548
      }
    ]
  },
  "libraryName": "LuisLibrary"
}
```

While the LUIS query returns something like this:

```json
{
  "query": "i want 20k rent 3 bedrooms",
  "topScoringIntent": {
    "intent": "SetFilters",
    "score": 0.999961853
  },
  "intents": [
    {
      "intent": "SetFilters",
      "score": 0.999961853
    },
    {
      "intent": "ChangeLocation",
      "score": 0.00164308527
    },
    {
      "intent": "None",
      "score": 0.000698036863
    },
    {
      "intent": "ShowProperties",
      "score": 0.0005648033
    },
    {
      "intent": "Settings",
      "score": 3.221991E-05
    },
    {
      "intent": "Notifications",
      "score": 9.207867E-10
    },
    {
      "intent": "ShowFavourites",
      "score": 2.942978E-12
    }
  ],
  "entities": [
    {
      "entity": "bedrooms",
      "type": "numericalFilters::minBedrooms",
      "startIndex": 18,
      "endIndex": 25,
      "score": 0.999940455
    },
    {
      "entity": "3 bedrooms",
      "type": "setNumerical",
      "startIndex": 16,
      "endIndex": 25,
      "score": 0.9976944
    },
    {
      "entity": "20k",
      "type": "builtin.number",
      "startIndex": 7,
      "endIndex": 9,
      "resolution": {
        "value": "20000"
      }
    },
    {
      "entity": "3",
      "type": "builtin.number",
      "startIndex": 16,
      "endIndex": 16,
      "resolution": {
        "value": "3"
      }
    },
    {
      "entity": "rent",
      "type": "listingType::rent",
      "startIndex": 11,
      "endIndex": 14,
      "score": 0.9967319
    },
    {
      "entity": "20k rent",
      "type": "setListingPrice",
      "startIndex": 7,
      "endIndex": 14,
      "score": 0.987298548
    }
  ],
  "compositeEntities": [
    {
      "parentType": "setNumerical",
      "value": "3 bedrooms",
      "children": [
        {
          "type": "builtin.number",
          "value": "3"
        },
        {
          "type": "numericalFilters::minBedrooms",
          "value": "bedrooms"
        }
      ]
    },
    {
      "parentType": "setListingPrice",
      "value": "20k rent",
      "children": [
        {
          "type": "builtin.number",
          "value": "20k"
        },
        {
          "type": "listingType::rent",
          "value": "rent"
        }
      ]
    }
  ]
}
```

So I [searched through the source code](https://github.com/Microsoft/BotBuilder/search?utf8=%E2%9C%93&q=compositeEntities&type=) and found out that there's no existing logic to handle `compositeEntities` attribute from LUIS query result in Node.js SDK but C# SDK does.

And I examined [`LuisRecognizer.ts`](https://github.com/Microsoft/BotBuilder/blob/master/Node/core/src/dialogs/LuisRecognizer.ts#L105) and validated my thought. After getting the LUIS query json in `recognize` function http callback, `compositeEntities` is currently ignored and not assigned to `ILuisResults`. Therefore we can't see `compositeEntities` in the dialog's intent.

Please kindly fix this asap as we need those composite entities to complete our NLP feature. Thanks.

**Comment by stevengum97** [comment]
Hi @josephktcheung, 

In regards to the composite entities not being passed to the client, they are being provided. Inside of the `entities` property for both of your provided JSON snippets both of your recognized composite entities exist, `'setNumerical'` and `'setListingPrice'`.

**Comment by josephktcheung** [comment]
Okay, but we'd like to have the composite entities attribute from the query so that we don't have to write custom logic to find the individual entities inside a composite entity

**Comment by josephktcheung** [comment]
I can't find a reason why `compositeEntities` cannot be provided in the intent while LUIS query has provided such information already.

**Comment by Stevenic** [comment]
I didn't realize this field was added but will get it added to the Node SDK. Sorry about that.

**Comment by stevengum97** [comment]
@Stevenic I created PR [**#2735**](https://github.com/Microsoft/BotBuilder/pull/2735) for this. I believe it covers the required changes and additions.

**Comment by nwhitmont** [comment]
resolved by #2735 PR (merged)
